models/classes/TestCategoryPresetProvider.php

```diff
@@ -18,78 +18,88 @@
  * Copyright (c) 2017 (original work) Open Assessment Technologies SA;
  */
 
+declare(strict_types=1);
+
 /**
  * @author Christophe Noël <christophe@taotesting.com>
  */
 
 namespace oat\taoQtiTest\models;
 
-
 use oat\oatbox\service\ConfigurableService;
 use oat\taoTests\models\runner\plugins\TestPluginService;
+use RuntimeException;
 
 class TestCategoryPresetProvider extends ConfigurableService
 {
-    const SERVICE_ID = 'taoQtiTest/CategoryPresetProvider';
+    public const SERVICE_ID = 'taoQtiTest/CategoryPresetProvider';
 
-    const GROUP_NAVIGATION  = 'navigation';
-    const GROUP_WARNING     = 'warning';
-    const GROUP_TOOLS       = 'tools';
+    public const GROUP_NAVIGATION = 'navigation';
+    public const GROUP_WARNING    = 'warning';
+    public const GROUP_TOOLS      = 'tools';
 
     private $allPresets;
 
+    private $isGroomed = false;
+
     /**
      * TestCategoryPresetProvider constructor.
+     *
      * @param array $options
      * @param array $allPresets - allow override of preset list
      */
-    public function __construct(array $options = [], $allPresets = [])
+    public function __construct(array $options = [], array $allPresets = [])
     {
         $this->allPresets = $allPresets;
 
         parent::__construct($options);
     }
 
-    protected function getPresetGroups()
+    /**
+     * @param string               $presetGroup
+     * @param TestCategoryPreset[] $presets
+     */
+    public function register(string $presetGroup, array $presets): void
     {
-        return [
-            self::GROUP_NAVIGATION => [
-                'groupId'    => self::GROUP_NAVIGATION,
-                'groupLabel' => __('Test Navigation'),
-                'groupOrder' => 100,
-                'presets'    => []
-            ],
-
-            self::GROUP_WARNING => [
-                'groupId'    => self::GROUP_WARNING,
-                'groupLabel' => __('Navigation Warnings'),
-                'groupOrder' => 200,
-                'presets'    => []
-            ],
+        if (!array_key_exists($presetGroup, $this->allPresets)) {
+            return;
+        }
 
-            self::GROUP_TOOLS => [
-                'groupId'    => self::GROUP_TOOLS,
-                'groupLabel' => __('Test-Taker Tools'),
-                'groupOrder' => 300,
-                'presets'    => []
-            ],
-        ];
+        foreach ($presets as $preset) {
+            /** @noinspection TypeUnsafeArraySearchInspection */
+            if (!in_array($preset, $this->allPresets[$presetGroup]['presets'])) {
+                $this->allPresets[$presetGroup]['presets'][] = $preset;
+            }
+        }
     }
 
     /**
      * Get all active presets
+     *
+     * @param bool $keepGroupKeys if `true` returns groups mapped to their group IDs
+     *
      * @return array - the sorted preset list
      */
-    public function getPresets()
+    public function getPresets(bool $keepGroupKeys = false): array
```

Comment:
It would technically be a breaking change if we started returning the results with keys here.
Somebody might've accessed a first one by a `[0]` index for example.